crypto/src/proofs/transparent.rs

```diff
@@ -587,6 +565,217 @@ impl TryFrom<transparent_proofs::SwapProof> for SwapProof {
     }
 }
 
+/// Transparent proof for swap creation.
+///
+/// Swaps create an output NFT encoding the swap data so they are most similar to Output operations.
+///
+/// This structure keeps track of the auxiliary (private) inputs.
+#[derive(Clone, Debug)]
+pub struct SwapProof {
+    // The diversified base for the destination address.
+    pub g_d: decaf377::Element,
+    // The transmission key for the destination address.
+    pub pk_d: ka::Public,
+    // The value of asset 1 in the swap.
+    pub value_t1: Value,
+    // The value of asset 2 in the swap.
+    pub value_t2: Value,
```

Comment:
Q: since the types of asset 1 and 2 are public, do we need to have `Value`s in the proof, or could we just have the integer amounts, and pass the asset IDs (or even the value generators) into the verification method as public inputs to the proof?

The reason this might matter is that we want the transparent proof to be an exact replica of the witness data for ZK proofs, so everything in the proof check has to turn into circuit constraints, and if we pass in the value generators, we could avoid having to perform the hash-to-group operation inside of the circuit.